server/service/src/requisition/common.rs

```diff
@@ -56,3 +58,43 @@ pub fn check_approval_status(requisition_row: &RequisitionRow) -> bool {
     }
     false
 }
+
+pub enum OrderTypeNotFoundError {
+    OrderTypeNotFound,
+    DatabaseError(RepositoryError),
+}
+
+pub fn check_emergency_order_within_max_items_limit(
+    connection: &StorageConnection,
+    program_id: &str,
+    order_type: &str,
+    requisition_lines: Vec<RequisitionLine>,
+) -> Result<(bool, i32), OrderTypeNotFoundError> {
+    let program_settings_ids = ProgramRequisitionSettingsRepository::new(connection)
+        .query(Some(ProgramRequisitionSettingsFilter::new().program(
+            ProgramFilter::new().id(EqualFilter::equal_to(program_id)),
+        )))?
+        .iter()
+        .map(|settings| settings.program_settings_row.id.clone())
+        .collect::<Vec<String>>();
+
+    let order_type = ProgramRequisitionOrderTypeRowRepository::new(connection)
+        .find_one_by_setting_and_name(&program_settings_ids, order_type)?
+        .ok_or(OrderTypeNotFoundError::OrderTypeNotFound)?;
+
+    let line_count = requisition_lines
+        .iter()
+        .filter(|line| line.requisition_line_row.requested_quantity != 0.0)
+        .count();
+
+    return Ok((
+        order_type.is_emergency && line_count < order_type.max_items_in_emergency_order as usize,
```

Comment:
I think should be `<=`

I'm getting this when requested is equal to the limit:

![Screenshot 2024-12-05 at 2 20 57 PM](https://github.com/user-attachments/assets/b578bc9a-9186-4ebf-b779-ab74f55868f4)

Reply:
This check is also currently being performed for both emergency and non emergency order types rather than just emergency orders!